packages/choice-input/src/ChoiceGroupMixin.js

```diff
@@ -0,0 +1,169 @@
+      super.addFormElement(child, indexToInsertAt);
+    }
+
+    /**
+     * @override from LionFieldset
+     */
+    // eslint-disable-next-line class-methods-use-this
+    get _childrenCanHaveSameName() {
+      return true;
+    }
+
+    /**
+     * @override from LionFieldset
+     */
+    // eslint-disable-next-line class-methods-use-this
+    get _childNamesCanBeDuplicate() {
+      return true;
+    }
+
+    _throwWhenInvalidChildModelValue(child) {
+      if (
+        typeof child.modelValue.checked !== 'boolean' ||
+        !Object.prototype.hasOwnProperty.call(child.modelValue, 'value')
+      ) {
+        throw new Error(
+          `The ${this.tagName.toLowerCase()} name="${
+            this.name
+          }" does not allow to register ${child.tagName.toLowerCase()} with .modelValue="${
+            child.modelValue
+          }" - The modelValue should represent an Object { value: "foo", checked: false }`,
+        );
+      }
+    }
+
+    _isEmpty() {
+      const value = this.modelValue;
+      if (this.multipleChoice) {
+        return this.modelValue.length === 0;
+      }
+
+      if (typeof value === 'string' && value === '') {
+        return true;
+      }
+      if (value === undefined || value === null) {
+        return true;
+      }
+      return false;
+    }
+
+    _checkSingleChoiceElements(ev) {
+      const { target } = ev;
+      if (target.checked === false) return;
+
+      const groupName = target.name;
+      this.formElementsArray
+        .filter(i => i.name === groupName)
+        .forEach(choice => {
+          if (choice !== target) {
+            choice.checked = false; // eslint-disable-line no-param-reassign
+          }
+        });
+      this.__triggerCheckedValueChanged();
+    }
+
+    _getCheckedElements() {
+      const filtered = this.formElementsArray.filter(el => el.checked === true);
+      if (this.multipleChoice) {
+        return filtered;
+      }
+      return filtered.length > 0 ? filtered[0] : undefined;
+    }
+
+    async _setCheckedElements(value, check) {
+      if (!this.__readyForRegistration) {
+        await this.registrationReady;
+      }
+
+      for (let i = 0; i < this.formElementsArray.length; i += 1) {
+        let currentElIsCheckedEl = false;
+        if (this.multipleChoice) {
+          currentElIsCheckedEl = value.includes(this.formElementsArray[i].value);
+        } else {
+          // Allows checking against custom values e.g. formattedValue or serializedValue
+          currentElIsCheckedEl = check(this.formElementsArray[i], value);
+        }
+
+        if (currentElIsCheckedEl) {
+          this.formElementsArray[i].checked = true;
+        } else if (this.multipleChoice) {
+          this.formElementsArray[i].checked = false;
+        }
+      }
+    }
+
+    __triggerCheckedValueChanged() {
+      const value = this.modelValue;
+      if (value != null && value !== this.__previousCheckedValue) {
+        this.touched = true;
```

Comment:
Where does this property come from? Could we make the relationships between mixins a bit more explicit in the current file?

Reply:
It's the same for FormRegistrarMixin. The ChoiceGroupMixin overrides / makes references to the FormRegistrarMixin. I know Thijs agrees with wrapping the mixin in the mixins it depends on and using the dedupemixin to dedupe. Feel free to experiment with that and see if it all works when you do it :). 'touched' property comes from InteractionStateMixin if I recall correctly.